### DISCUSSION TITLE
Behavior如何写到Style中去

### DISCUSSION THREAD

**Discussion by ligu1983** [BYJRK/DotNet-Discussions#263] [WPF 开发]
<TextBox Text="{Binding DataList[2], Mode=OneWayToSource, FallbackValue=0}">
    <i:Interaction.Behaviors>
        <behavior:NumericRangeBehavior MaxValue="90" MinValue="-90" />
    </i:Interaction.Behaviors>
    <i:Interaction.Triggers>
        <i:EventTrigger EventName="LostFocus">
            <i:InvokeCommandAction Command="{Binding SendParaCmd}">
                <i:InvokeCommandAction.CommandParameter>
                    <MultiBinding Converter="{StaticResource MultiBindingConverter}">
                        <Binding Source="2" />
                        <Binding Path="Text" RelativeSource="{RelativeSource AncestorType=TextBox}" />
                    </MultiBinding>
                </i:InvokeCommandAction.CommandParameter>
            </i:InvokeCommandAction>
        </i:EventTrigger>
    </i:Interaction.Triggers>
</TextBox>
我界面上有多个这样的TextBox，都会用到同样的behavior和trigger，因此我想把这部分代码写到一个style中，因此我这样写的时候报错了。
<Setter Property="i:Interaction.Behaviors">
    <Setter.Value>
        <behavior:NumericRangeBehavior MaxValue="90" MinValue="-90" />
    </Setter.Value>
</Setter>
IDE提示Setter.Value应该是一个BehaviorCollection，但是我不知道怎么添加这个BehaviorCollection，求大神赐教。

**Comment by BYJRK** [comment]
基本做不到，建议放弃。如果有批量添加行为的需求，或许可以考虑使用模板。
另外，也不是完全做不到，但是很复杂。这里有一个[回答](https://stackoverflow.com/questions/1647815/how-to-add-a-blend-behavior-in-a-style-setter/4779168#4779168)，仅供参考

**Reply by ligu1983** [reply]
非常感谢。

**Reply by BYJRK** [reply]
还有一种方式，就是写一个非常简单的自定义控件，比如直接继承 `TextBox`，样式也直接使用 `TextBox`（这部分可以了解一下自定义控件的写法，以及默认样式的指定方式），然后可以考虑在构造或者 `Loaded` 事件中，也就是在代码后台去手动添加响应的 `Behavior`
